lib/heuristicBlocker.js

```diff
@@ -331,12 +333,31 @@ let getChannelInfo = function(channel, win) {
  * @return Boolean Whether this request appears to be tracking the user
  */
 let hasTracking = function(channel, channelInfo, cookies) {
+  let tracking = hasCookieTracking(channel, channelInfo, cookies) ||
+    hasSupercookieTracking(channel, channelInfo) ||
+    hasCanvasFingerprinting(channel, channelInfo);
 
+  return tracking;
+
+};
+
+var hasSupercookieTracking = function(channel, channelInfo){
+  return settingsMap.supercookies && 
+    !!settingsMap.supercookies[channelInfo.origin];
+};
+
+var hasCanvasFingerprinting = function(channel, channelInfo){
+  return settingsMap.canvasTracking && 
+    settingsMap.canvasTracking[channelInfo.origin] &&
+    !!settingsMap.canvasTracking[channelInfo.origin].fingerprinting;
+};
+
+var hasCookieTracking = function(channel, channelInfo, cookies){
   if (!cookies || Object.getOwnPropertyNames(cookies).length < 1) {
     return false;
   }
 
-  let { origin, parentOrigin } = channelInfo;
```

Comment:
I only removed getting parentOrigin since we don't use it in this function, we are still getting origin